neutron_plugin/port.py

```diff
@@ -127,3 +158,28 @@ def connect_security_group(neutron_client, **kwargs):
 @with_neutron_client
 def creation_validation(neutron_client, **kwargs):
     validate_resource(ctx, neutron_client, PORT_OPENSTACK_TYPE)
+
+
+def _get_server_floating_ip(neutron_client, server_id):
```

Comment:
it'd probably have been better to iterate the specific server's network-IPs rather than over all floating IPs - simply in the sense that iterating over all floating IPs is in a "global context" and there might be many of those, as opposed to the local context of IPs of a given server.